packages/integration-objects/test/ValidatePrimaryFieldsForRates.test.js

```diff
@@ -0,0 +1,187 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+const { validatePrimaryKeyFieldsForCurrencyExchangeRate } = require('../srv/validations/ValidatePrimaryFieldsForRates');
+const ErrorStatuses = require('../srv/utils/ErrorStatuses');
+const RateExtensionConstants = require('../srv/utils/RateExtensionConstants');
+
+describe('validate valid fields among primary keys', function () {
+  it('does not return any error- primary keys validation', function () {
+    const reqObj = {
+      tenantID: 'tenant1',
+      dataProviderCode: 'TW',
+      dataSource: 'THR',
+      exchangeRateType: 'aA',
+      fromCurrencyThreeLetterISOCode: 'CND',
+      toCurrencyThreeLetterISOCode: 'INR',
+      validFromDateTime: '2020-02-28T06:38:29Z',
+      exchangeRateValue: '80.00'
+    };
+    expect(validatePrimaryKeyFieldsForCurrencyExchangeRate(reqObj)).toBeUndefined();
+  });
+
+  it('dataProviderCode null check', () => {
+    const reqObj = {
+      tenantID: 'tenant1',
+      dataProviderCode: null,
+      dataSource: 'THR',
+      exchangeRateType: 'aA',
+      fromCurrencyThreeLetterISOCode: 'AED',
+      toCurrencyThreeLetterISOCode: 'INR',
+      validFromDateTime: '2020-02-28T06:38:29Z',
+      exchangeRateValue: '80.00'
+    };
+    try {
+      validatePrimaryKeyFieldsForCurrencyExchangeRate(reqObj);
+    } catch (err) {
+      expect(err.message).toBe('Provide a valid value for dataProviderCode. The value must be 1 - 15 characters long.');
```

Comment:
Since it is shorter either use the `expect(XYZ).toThrowError(yourError)` or if you do not want to have the actual assertion in your code you can even use the `expect(XYZ).toThrowMatchingSnapshot()`. Also not that this works for async function than you use await expect(asyncFunction).rejects.toThrow(). Check the docu for details in the async case I am not 100% sure, perhaps you need to wrap the function.

Reply:
This is fixed.